### ISSUE TITLE
EKS list Kubernetes versions

### ISSUE THREAD

**Issue by ewbankkit** [hashicorp/terraform-provider-aws#40706]
> Amazon Elastic Kubernetes Service (EKS) now lets you programmatically access availability of Kubernetes versions.

* (https://aws.amazon.com/about-aws/whats-new/2024/12/amazon-eks-access-kubernetes-version-availability/)[Announcement]

### New Data Source(s)

* `aws_eks_cluster_versions`
* `aws_eks_cluster_version`

Requires AWS SDK for Go v2 `Release 2024-12-23`: https://github.com/hashicorp/terraform-provider-aws/pull/40687.

**Comment by hjoshi123** [comment]
@ewbankkit Can I work on this?

**Comment by hjoshi123** [comment]
@ewbankkit we cant work on this unless the aws sdk upgrade is merged right?

**Comment by hjoshi123** [comment]
@ewbankkit is it okay to upgrade the deps to work on this task and we can merge the main into this once the deps are upgraded in main?

**Comment by ewbankkit** [comment]
@hjoshi123 Yes, as we are waiting on a resolution to https://github.com/aws/aws-sdk-go-v2/issues/2939 before we can merge the full set of AWS SDK for Go v2 updates, I suggest `go get github.com/aws/aws-sdk-go-v2/service/eks && go mod tidy` to take just the EKS upgrade.

**Comment by hjoshi123** [comment]
Yup that's what I thought too. Thanks for confirming @ewbankkit

**Comment by hjoshi123** [comment]
@ewbankkit I am done with implementing `aws_eks_cluster_versions` data source was wondering if we need `aws_eks_cluster_version` data source since the API always returns an array and it doesnt give any unique identifier to query to get an individual result out of it. For example this is how the command looks like 

```
 aws eks describe-cluster-versions --cluster-version=1.31
{
    "clusterVersions": [
        {
            "clusterVersion": "1.31",
            "clusterType": "eks",
            "defaultPlatformVersion": "eks.16",
            "defaultVersion": true,
            "releaseDate": "2024-09-25T18:00:00-06:00",
            "endOfStandardSupportDate": "2025-11-25T17:00:00-07:00",
            "endOfExtendedSupportDate": "2026-11-25T17:00:00-07:00",
            "status": "STANDARD_SUPPORT",
            "kubernetesPatchVersion": "1.31.4"
        }
    ]
}
```

So I was wondering what was the reasoning behind having another data source?

**Comment by hjoshi123** [comment]
Also, @ewbankkit might have found another bug on the upstream sdk.. the CLI and the Call to DescribeClusterVersions API returns the status as             "status": "STANDARD_SUPPORT" or EXTENDED_SUPPORT however, the enum types on their SDK is mapped to the below which makes the acceptance test fail
```
ClusterVersionStatusUnsupported     ClusterVersionStatus = "unsupported"
	ClusterVersionStatusStandardSupport ClusterVersionStatus = "standard-support"
	ClusterVersionStatusExtendedSupport ClusterVersionStatus = "extended-support"
```

I am not sure what to do. Is this something we have to raise an issue on the aws sdk side?

**Comment by hjoshi123** [comment]
Raised the following PR on upstream https://github.com/aws/aws-sdk-go-v2/issues/2942
